src/DotVVM.Framework.Hosting.Owin/AppBuilderExtensions.cs

```diff
@@ -26,27 +26,53 @@ public static class AppBuilderExtensions
         /// in production.
         /// </param>
         /// <param name="debug">A value indicating whether the application should run in debug mode.</param>
-        /// <param name="options">An action to configure DotVVM services and extensions.</param>
-        public static DotvvmConfiguration UseDotVVM<TStartup>(this IAppBuilder app, string applicationRootPath, bool useErrorPages = true, bool debug = true, Action<IDotvvmOptions> options = null)
+        public static DotvvmConfiguration UseDotVVM<TStartup, TServiceConfigurator>(this IAppBuilder app, string applicationRootPath, bool useErrorPages = true, bool debug = true)
             where TStartup : IDotvvmStartup, new()
+            where TServiceConfigurator : IDotvvmServiceConfigurator, new()
         {
-            var config = app.UseDotVVM(applicationRootPath, useErrorPages, debug, options);
+            var serviceConfigurator = new TServiceConfigurator();
+            var config = app.UseDotVVM(applicationRootPath, useErrorPages, debug, serviceConfigurator.ConfigureServices);
             new TStartup().Configure(config, applicationRootPath);
             return config;
         }
 
-        private static DotvvmConfiguration UseDotVVM(this IAppBuilder app, string applicationRootPath, bool useErrorPages, bool debug, Action<IDotvvmOptions> options)
+        /// <summary>
+        /// Adds DotVVM to the <see cref="IAppBuilder" /> request execution pipeline.
+        /// </summary>
+        /// <param name="app">The <see cref="IAppBuilder" /> instance.</param>
+        /// <param name="applicationRootPath">The path to application's root directory. It is used to resolve paths to views, etc.</param>
+        /// <param name="useErrorPages">
+        /// A value indicating whether to show detailed error page if an exception occurs. Disable this
+        /// in production.
+        /// </param>
+        /// <param name="debug">A value indicating whether the application should run in debug mode.</param>
+        public static DotvvmConfiguration UseDotVVM<TStartup>(this IAppBuilder app, string applicationRootPath, bool useErrorPages = true, bool debug = true)
+            where TStartup : IDotvvmStartup, new()
         {
-            var config = DotvvmConfiguration.CreateDefault(s =>
+            var startup = new TStartup();
+            Action<IDotvvmServiceCollection> options = service => { };
```

Comment:
It would be better to pass an implementation of `IDotvvmServiceConfigurator` or null to the private `UseDotVVM` method. Or at least avoid allocating an empty delegate here.